icu4c/source/i18n/formattedvalue.cpp

```diff
@@ -193,6 +193,9 @@ ucfpos_close(UConstrainedFieldPosition* ptr) {
 }
 
 
+#if defined(__clang__) || U_GCC_MAJOR_MINOR >= 1100
```

Comment:
I don't think we should suppress the warning. Warning is our friend who watch our back when we accidentially did the wrong thing. I think a simple fix is just
```
iff --git a/icu4c/source/i18n/formattedvalue.cpp b/icu4c/source/i18n/formattedvalue.cpp
index 1030661f22..a904b78a54 100644
--- a/icu4c/source/i18n/formattedvalue.cpp
+++ b/icu4c/source/i18n/formattedvalue.cpp
@@ -211,7 +211,7 @@ ufmtval_getString(
     }
     // Note: this line triggers -Wreturn-local-addr, but it is safe because toTempString is
     // defined to return memory owned by the ufmtval argument.
-    return readOnlyAlias.getBuffer();
+    return impl->fFormattedValue->toTempString(*ec).getBuffer();
 }
 
```